Quorum/entry_points/setup_quorum.py

```diff
@@ -0,0 +1,69 @@
+import shutil
+import argparse
+from pathlib import Path
+
+import Quorum.utils.pretty_printer as pp
+
+
+def get_working_directory() -> Path:
+    parser = argparse.ArgumentParser(description="Setup Quorum project.")
+    parser.add_argument(
+        '--working_dir',
+        default=Path.cwd() / 'quorum_project',
+        type=Path,
+        help="Directory to set up the Quorum project."
+    )
+    args = parser.parse_args()
+    return args.working_dir
+
+
+def setup_quorum(working_dir: Path):
+    """
+    Initializes Quorum environment by copying template files to the specified directory.
+
+    Args:
+        working_dir (Path): Target directory for setting up Quorum.
+
+    Raises:
+        shutil.Error: If copying files fails.
+        OSError: If directory creation fails.
+    """
+    templates_dir = Path(__file__).parent.parent / 'templates'
+    target_dir = working_dir.resolve()
+
+    if not target_dir.exists():
+        pp.pretty_print(f"Creating directory: {target_dir}", pp.Colors.INFO)
+        target_dir.mkdir(parents=True, exist_ok=True)
+
+    # Collect all file names to copy from the templates directory
+    template_files = [f.name for f in templates_dir.iterdir() if f.is_file()]
+
+    for file_name in template_files:
+        src = templates_dir / file_name
+        dest = target_dir / '.env' if file_name == '.env.example' else target_dir / file_name
+
+        if dest.exists():
+            pp.pretty_print(f"File exists: {dest}. Skipping.", pp.Colors.WARNING)
+            continue
+
+        shutil.copy(src, dest)
+        pp.pretty_print(f"Copied {file_name} to {dest}", pp.Colors.SUCCESS)
```

Comment:
```suggestion
    template_files = [f for f in templates_dir.iterdir() if f.is_file()]

    for file in template_files:
        dest = target_dir / '.env' if file.name == '.env.example' else target_dir / file.name

        if dest.exists():
            pp.pretty_print(f"File exists: {dest}. Skipping.", pp.Colors.WARNING)
            continue

        shutil.copy(file, dest)
        pp.pretty_print(f"Copied {file.name} to {dest}", pp.Colors.SUCCESS)
```

Last one since functionality changed

Reply:
Not sure its going to work for the .env check. so ill keep it for now

Reply:
It will work as you iterate on absolute files